### PR TITLE
Added a class that implements read only of a file.

### PR DESCRIPTION
**Pull request by rohe**
The file is expected to contain a set of information pieces. One per line. The class acts like a list, except that it's non-mutable..

### CODE REVIEW

File: tests/test_14_read_only_list_file.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        os.unlink(FILE_NAME)
+        os.unlink(f"{FILE_NAME}.lock")
+
+    _read_only = ReadOnlyListFile(FILE_NAME)
+    assert len(_read_only) == 0
+
+    with open(FILE_NAME, "w") as fp:
+        for item in ["one", "two", "three"]:
+            fp.write(item)
+            fp.write("\n")
```
Comment (peppelinux) [inline tests/test_14_read_only_list_file.py]:
```suggestion
        fb.writelines(["one", "two", "three"])
```

File: src/idpyoidc/storage/listfile.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            try:
+                lock = FileLock(f"{fname}.lock")
+                with lock:
+                    fp = open(fname, "r")
+                    info = [x.strip() for x in fp.readlines()]
+                lock.release()
+                if info == "":
+                    return None
+                else:
+                    return info
```
Comment (peppelinux) [inline src/idpyoidc/storage/listfile.py]:
```suggestion
            return info or None
```

**Review by peppelinux — APPROVED** [review]
few hints, approved

File: tests/test_14_read_only_list_file.py (hunk 1 above)
Comment (rohe) [reply tests/test_14_read_only_list_file.py]: The '\n' at the end of the line is significant and writelines doesn't add those so I adopted a variant of what you proposed.

File: src/idpyoidc/storage/listfile.py (hunk 2 above)
Comment (rohe) [reply src/idpyoidc/storage/listfile.py]: OK

**Pull request merged**